lib/evmone/baseline.cpp

```diff
@@ -271,12 +273,37 @@ evmc_result execute(const VM& vm, ExecutionState& state, const CodeAnalysis& ana
 }
 
 evmc_result execute(evmc_vm* c_vm, const evmc_host_interface* host, evmc_host_context* ctx,
-    evmc_revision rev, const evmc_message* msg, const uint8_t* code, size_t code_size) noexcept
+    evmc_revision rev, const evmc_message* msg, const uint8_t* code, size_t code_size, unsigned int sample_size) noexcept
 {
     auto vm = static_cast<VM*>(c_vm);
-    const auto jumpdest_map = analyze({code, code_size});
-    auto state =
-        std::make_unique<ExecutionState>(*msg, rev, *host, ctx, bytes_view{code, code_size});
-    return execute(*vm, *state, jumpdest_map);
+
+    evmc_result result;
+
+    // gas-cost-estimator decoration
+    //
+    for (unsigned int run_id = 0; run_id < sample_size; ++run_id) {
+        auto state =
```

Comment:
You should allocate it outside of the loop. Inside the loop use `state->reset(...)`.

Reply:
as discussed - it's better to `reset`, but not a big error to realloc. I'll make a follow-up issue out of this